doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+            "namespace": "org.kaaproject.kaa.schema.example",
+            "fields": [
+                {
+                    "name": "samplePeriod",
+                    "type": "int",
+                    "by_default": "1"
+                }
+            ]
+        }
+1. Open the admin UI in your browser and log in as
+    a tenant developer (user/password: devuser/devuser123).
+1. Open the relevant **Log schemas** window (**Applications =>
+    My First Kaa Application => Schemas => Log**) and click **Add schema**.
+1. In the **Add log schema** window, enter the name and description
+    of the new data collection schema.
+
+    ![Data collection scheme](attach/new_data_schema.png)
+1. Scroll down and use the **Upload from file** function to find the previously
+    created json file with the schema. Alternatively, you can use the
+    [Schema Avro UI]({{root_url}}/Administration-guide/Tenants-and-applications-management/#avro-ui-forms).
+    form to create the schema.
+1. Click **Upload**.
+1. Click **Add** at the top of the window.
+1. Repeat uploading and adding for the configuration schema
+    (*configuration-schema.json*) scheme:
+
+    ![Data collection scheme](attach/new_config_schema.png)
+
+As a result of these operations you will see configuration and data collections
+schemes in the list:
+![Data collection schema complete](attach/log_schema_list.png)
+![Configuraion schema complete](attach/config_schema_list.png)
+
+In this screenshot, version 2.0 is log and configuraion schema version that
+was just created. We will use this version for the SDK generation later.
+
+# Log appenders setup
+
+In order to use Data Collection feature it is required to setup **Log Appender**.
+In this example we will use MongoDB log appender.
+Refer to [Admin UI section describing how to setup it]({{root_url}}/Administration-guide/Tenants-and-applications-management/#mongodb-log-appender).
+
+# Generating SDK
+
+To generate an SDK for the new application, proceed as follows:
+
+1. Select the **My First Kaa Application** application and click **Generate SDK**.
+
+    ![SDK generation](attach/generate_sdk.png)
+1. Click **Add SDK profile**.
+
+    ![Add SDK](attach/add_sdk.png)
+1. In the **Add SDK profile** window, fill in the fields as shown in the following
+screenshot  and then click **Add**.
+
+    ![Configure SDK](attach/configure_sdk.png)
+1. Now the SDK is configured and ready to be generated. Click **Generate SDK** for
+corresponding SDK profile. In the **Generate SDK** window select the target platform for
+your SDK and click Generate SDK.
+
+    ![Generate SDK](attach/generate_configured_sdk.png)
+
+    ![Generate SDK](attach/select_target_platform.png)
+
+After the SDK is generated, you will be presented with a window asking you to save a
+.jar file with the generated SDK (for Java) or an archive with the generated SDK (for C,
+C++ or Objective-C). Specify the file name and location on your computer and then click
```

Comment:
Please try avoiding specific information regarding the file types. This will never be updated again and thus will cause inconsistency should that things change.